libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/ranker/TopKDataLayoutCandidateSelector.java

```diff
@@ -0,0 +1,27 @@
+package com.linkedin.openhouse.datalayout.ranker;
+
+import com.linkedin.openhouse.datalayout.strategy.DataLayoutStrategy;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.PriorityQueue;
+import org.apache.commons.lang3.tuple.Pair;
+
+/** A greedy candidate selector that selects the top K data layout strategies, ranked by scores. */
+public class TopKDataLayoutCandidateSelector extends BaseDataLayoutCandidateSelector {
```

Comment:
Is it the same as greedy where the cost budget is unlimited? Maybe we can extend the greedy one to handle inf budget

Reply:
True. Let me remove this.

Reply:
done